src/coreclr/jit/gentree.cpp

```diff
@@ -26150,6 +26150,21 @@ bool GenTreeLclVar::IsNeverNegative(Compiler* comp) const
     return comp->lvaGetDesc(GetLclNum())->IsNeverNegative();
 }
 
+//------------------------------------------------------------------------
+// GenTreeLclVar::IsSpan: Gets true if the lcl var is a Span<T>; otherwise false.
+//
+// Arguments:
+//    comp - the compiler instance
+//
+// Return Value:
+//    true if the lcl var is a Span<T>; otherwise false.
+//
+bool GenTreeLclVar::IsSpan(Compiler* comp) const
+{
+    assert(OperIs(GT_LCL_VAR));
+    return comp->lvaGetDesc(GetLclNum())->IsSpan();
+}
```

Comment:
I would remove this and just check the property on `LclVarDsc` directly in the two places we are using it. Mainly it looks a bit weird to be because it will return true for the implicit byref `LCL_VARs` which really are pointers to spans.